pkgs/os-specific/linux/device-tree.nix

```diff
@@ -0,0 +1,28 @@
+{ stdenvNoCC, dtc, findutils, raspberrypifw }:
+
+with stdenvNoCC.lib; {
+  applyOverlays = (base: overlays: stdenvNoCC.mkDerivation {
+    name = "device-tree-overlays";
+    nativeBuildInputs = [ dtc findutils ];
+    buildCommand = let
+      quotedDtbos = concatMapStringsSep " " (o: "\"${toString o}\"") (toList overlays);
+    in ''
+      for dtb in $(find ${base} -name "*.dtb" ); do
+        outDtb=$out/$(realpath --relative-to "${base}" "$dtb")
+        mkdir -p "$(dirname "$outDtb")"
+        fdtoverlay -o "$outDtb" -i "$dtb" ${quotedDtbos};
+      done
+    '';
+  });
+
+  raspberryPiDtbs = stdenvNoCC.mkDerivation {
```

Comment:
Are you sure you want to put the raspberry pi overlays here? This package could grow a bit large if you start adding overlays for other boards.

Reply:
Nope, I'm not at all sure. Where would you suggest they go?

Reply:
No, I'm not at all sure. Where would you suggest they go?

Reply:
I would probably put them in separate nix files in pkgs/os-specific/linux/overlays (or something like that), but I'd like a second opinion before I suggest you do that.

Reply:
I've moved them - does it look better now?